src/lib/logger.js

```diff
@@ -0,0 +1,18 @@
+const chalk = require('chalk');
+
+/**
+ * Logger with `log` and `error` function.
+ */
+module.exports = {
+  log(...args) {
+    const [format, ...rest] = args;
+    console.log(`${chalk.grey('[Semantic release]:')} ${format}`, ...rest.map(arg => chalk.magenta(arg)));
+  },
+  error(...args) {
+    const [format, ...rest] = args;
+    console.error(
+      `${chalk.grey('[Semantic release]:')} ${chalk.red(format instanceof Error ? format.stack : format)}`,
+      ...rest.map(arg => chalk.red(arg instanceof Error ? arg.stack : arg))
```

Comment:
why only log `arg.stack`? Afaik `console.error()` should log the stack of errors, but if the error has more metadata it will log that too. For example, child_process.exec() errors have an `stderr` property that gives you the actual error output of the command that failed, without this the error is often times useless

Reply:
I checked that `arg.stack` logs the error type and the stacktrace. Can you confirm it won't log any other parameters?

This log is useful for debugging purpose. Is there a scenario in which the stacktrace is not enough to find where the problem happen and reproduce?
For example is the case of `child_process.exec()` with the stacktrace you could find the line in the source that caused the error and see the command executed there.

On a side note a plugin can also catch its own errors and log them before re-throwing them to `semantic-release`.

Reply:
This is what a logged child process `err.stack` looks like:

```
Error: Command failed: git status
    at checkExecSyncError (child_process.js:591:13)
    at Object.execSync (child_process.js:631:13)
    at repl:1:32
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7)
```

This is what `console.error(err)` looks like:
```
{ Error: Command failed: git status
fatal: Not a git repository (or any of the parent directories): .git

    at checkExecSyncError (child_process.js:591:13)
    at Object.execSync (child_process.js:631:13)
    at repl:1:32
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7)
  error: null,
  cmd: 'git status',
  file: '/bin/sh',
  args: [Array],
  options: [Object],
  envPairs: [Array],
  stderr: 'fatal: Not a git repository (or any of the parent directories): .git\n',
  stdout: '',
  pid: 1492,
  output: [Array],
  signal: null,
  status: 128 }
```
The former log doesn't help, the latter does because the important metadata (`stderr`) is included.

This was an actual real life situation: https://github.com/semantic-release/semantic-release/issues/477
Took me forever to find out what went wrong

This is just an example, it's quite common to attach more metadata to properties, e.g. when wrapping (an)other error(s)
```js
throw Object.assign(new Error('Multiple other operations failed'), { errors })
```

Reply:
Ok make sense. I'll fix it shortly. Thanks for the feedback !